src/runtime/threads/thread_data.cpp

```diff
@@ -35,56 +35,17 @@ namespace hpx { namespace threads
     void intrusive_ptr_release(thread_data_base* p)
     {
         if (0 == --p->count_)
-            delete p;
-    }
-
-    ///////////////////////////////////////////////////////////////////////////
-    void* thread_data::operator new(std::size_t size, thread_pool& pool)
-    {
-        HPX_ASSERT(sizeof(thread_data) == size);
-
-        void *ret = reinterpret_cast<void*>(pool.allocate());
-        if (0 == ret)
-        {
-            HPX_THROW_EXCEPTION(out_of_memory,
-                "thread_data::operator new",
-                "could not allocate memory for thread_data");
-        }
-
-#ifdef HPX_DEBUG_THREAD_POOL
-        using namespace std;    // some systems have memset in namespace std
-        memset (ret, initial_value, sizeof(thread_data));
-#endif
-        return ret;
-    }
-
-    void thread_data::operator delete(void *p, std::size_t size)
-    {
-        HPX_ASSERT(sizeof(thread_data) == size);
-
-        if (0 != p)
         {
-            thread_data* pt = static_cast<thread_data*>(p);
-            thread_pool* pool = pt->pool_;
-            HPX_ASSERT(pool);
-
-#ifdef HPX_DEBUG_THREAD_POOL
-            using namespace std;    // some systems have memset in namespace std
-            memset (static_cast<void*>(pt), freed_value, sizeof(thread_data)); //-V598
-#endif
-            pool->deallocate(pt);
-        }
-    }
-
-    void thread_data::operator delete(void *p, thread_pool& pool)
-    {
-        if (0 != p)
-        {
-#ifdef HPX_DEBUG_THREAD_POOL
-            using namespace std;    // some systems have memset in namespace std
-            memset (p, freed_value, sizeof(thread_data));
-#endif
-            pool.deallocate(static_cast<thread_data*>(p));
+            if (thread_data* td = dynamic_cast<thread_data*>(p))
```

Comment:
@hkaiser sorry, Hartmut, don't get your point. This operator delete doesn't exist in this PR (I removed it). global delete is needed to be called for stackless_thread_data, as its object is allocated with global new here https://github.com/STEllAR-GROUP/hpx/blob/fixing_1763_2/hpx/runtime/threads/policies/thread_queue.hpp#L220